php/api/controller/class-audit-posts-controller.php

```diff
@@ -677,8 +677,10 @@ public function check_read_permission( $post ) {
 		 * Still allowed users with the right capabilities to read this post.
 		 */
 		$visibility   = get_post_meta( $post->ID, 'visibility', true );
-		$current_user = get_current_user_id();
-		if ( 'public' !== $visibility && $post->post_author !== $current_user && ! current_user_can( 'read_private_posts' ) ) {
+		$current_user = wp_get_current_user();
+		$is_author    = $post->post_author !== $current_user->ID && in_array( 'api_client', (array) $current_user->roles );
```

Comment:
@rheinardkorf I would change it to `$post->post_author === $current_user->ID` and then below do a `! $is_author` check instead as the name implies a true value.